# The target sensor frame.
# AP20 do not have orientation so if we want to compare against that we have to convert to its sensor frame.
target_frame: prism
filter_by_covariance: False
covariance_percentile_threshold: 95 #percentile
#
remove_until_motion: True
distance_threshold: 1 #distance in meters
#
ap20_peak_rejection: True
ap20_peak_rejection_threshold: 0.1 #second
ap20_peak_rejection_trailing_window: 5 #trailing indices to remove
#
project_to_plane: "xyz" #xy or xyz

# If topic entry is left empty "", the bag only going to be used for /tf & /tf_static reading and not for topic reading
bag_topic_pairs:
  # - bag: _gt_pose_live.bag
  #   test_name: "gmsf_live"
  #   topic: /gt_box/ground_truth/pose_with_covariance
  #   frame: box_base
  #   evaluations:
  #     parameters:
  #       t_max_diff: 0.005
  #       t_offset: 0.0
  #       n_to_align: -1
  #       correct_scale: False
  #       delta: 1
  #       unit: "m" #m or frame
  #     ape_rpe:
  #       reference_file: ""
  #     point_relation:
  #       reference_file: "ap20_robot"

  - bag: _gt_pose_optimized.bag
    test_name: "gmsf_optimized"
    topic: /gt_box/ground_truth/pose_with_covariance
    frame: box_base
    evaluations:
      parameters:
        t_max_diff: 0.005
        t_offset: 0.0
        n_to_align: -1
        correct_scale: False
        delta: 1
        unit: "m" #m or frame
      ape_rpe:
        reference_file: ""
      point_relation:
        reference_file: "ap20_robot"

  # - bag: _gt_pose_no_gnss.bag
  #   test_name: "gmsf_no_gnss"
  #   topic: /gt_box/ground_truth/pose_with_covariance
  #   frame: box_base
  #   evaluations:
  #     parameters:
  #       t_max_diff: 0.005
  #       t_offset: 0.0
  #       n_to_align: -1
  #       correct_scale: False
  #       delta: 1
  #       unit: "m" #m or frame
  #     ape_rpe:
  #       reference_file: ""
  #     point_relation:
  #       reference_file: "ap20_robot"

  #
  - bag: _cpt7_ie_tc.bag
    test_name: "offline_ie_tc"
    topic: /gt_box/inertial_explorer/tc/gt_poses_novatel
    frame: box_base
    evaluations:
      parameters:
        t_max_diff: 0.005
        t_offset: 0.0
        n_to_align: -1
        correct_scale: False
        delta: 1
        unit: "m" #m or frame
      ape_rpe:
        reference_file: ""
      point_relation:
        reference_file: "ap20_robot"

  # - bag: _cpt7_ie_lc.bag
  #   test_name: "offline_ie_lc"
  #   topic: /gt_box/inertial_explorer/lc/gt_poses_novatel
  #   frame: box_base
  #   evaluations:
  #     parameters:
  #       t_max_diff: 0.005
  #       t_offset: 0.0
  #       n_to_align: -1
  #       correct_scale: False
  #       delta: 1
  #       unit: "m" #m or frame
  #     ape_rpe:
  #       reference_file: ""
  #     point_relation:
  #       reference_file: "ap20_robot"

  # - bag: _cpt7_ie_ppp.bag
  #   test_name: "offline_ie_ppp"
  #   topic: /gt_box/inertial_explorer/ppp/gt_poses_novatel
  #   frame: box_base
  #   evaluations:
  #     parameters:
  #       t_max_diff: 0.005
  #       t_offset: 0.0
  #       n_to_align: -1
  #       correct_scale: False
  #       delta: 1
  #       unit: "m" #m or frame
  #     ape_rpe:
  #       reference_file: ""
  #     point_relation:
  #       reference_file: "ap20_robot"

    #
  - bag: _jetson_ap20_robot.bag
    test_name: "ap20_robot"
    topic: "/gt_box/ap20/prism_position"
    frame: "prism"
    evaluations:
      parameters:
        t_max_diff: 0.025
        t_offset: 0.0
        n_to_align: -1
        correct_scale: False
        delta: 1
        unit: "m" #m or frame
      ape_rpe:
        reference_file: ""
      point_relation:
        reference_file: "ap20_robot"

    #
  - bag: _tf_static.bag
    test_name: ""
    topic: ""
    frame: ""